libdnf/utils/temp.cpp

```diff
@@ -0,0 +1,33 @@
+#include "temp.hpp"
+
+#include <cstdlib>
+
+
+namespace libdnf::utils {
+
+
+TempDir::TempDir(const std::string & prefix) {
+    // get path to the system temp directory
+    auto temp = std::filesystem::temp_directory_path();
+
+    // append /<prefix>XXXXXX as requested in mkdtemp(3)
+    temp /= prefix + "XXXXXX";
+    path = mkdtemp(const_cast<char *>(temp.native().c_str()));
+}
+
+
+TempDir::TempDir(const std::string & prefix, const std::vector<std::string> & subdirs)
+    : TempDir(prefix) {
+    for (auto & subdir : subdirs) {
+        std::filesystem::create_directory(path / subdir);
+    }
+}
+
+
+TempDir::~TempDir() {
+    // remove the temp directory and all its content
+    std::filesystem::remove_all(path);
```

Comment:
`std::filesystem::remove_all` may throw an exception.
And uncaught exception in the destructor is not a good idea.

Reply:
exception caught